fiftyone/utils/labels.py

```diff
@@ -673,3 +674,116 @@ def classifications_to_detections(
                 detections.append(detection)
 
             image[out_field] = fol.Detections(detections=detections)
+
+
+def perform_nms(
+    sample_collection,
+    in_field,
+    out_field,
+    conf_threshold=0.6,
+    iou_threshold=0.5,
+    progress=None,
+):
+    """Performs Non-Maximum Suppression (NMS) on the
+    :class:`fiftyone.core.labels.Detections` field containing detections.
+
+    NMS is a post-processing technique used in object detection to eliminate
+    duplicate detections and select the most relevant detected objects. This
+    helps reduce false positives.
+
+    Args:
+        sample_collection: a
+            :class:`fiftyone.core.collections.SampleCollection`
+        in_field: the name of the :class:`fiftyone.core.labels.Detections` field
+        out_field: the name of the :class:`fiftyone.core.labels.Detections`
+            field to populate. Creates a new field if one doesn't already exist
+            with this name.
+        conf_threshold (0.6): a floating-point value between 0 and 1 representing
+            the minimum confidence score required for a detection to be considered
+            valid. Detections with confidence scores lower than this threshold
+            will be discarded during the Non-Maximum Suppression (NMS) process.
+        iou_threshold (0.5): a floating-point value between 0 and 1 representing
+            the Intersection over Union (IoU) threshold used in the NMS algorithm.
+            It determines the minimum overlap required between bounding boxes for
+            them to be considered duplicates. Bounding boxes with IoU values
+            greater than or equal to this threshold will be suppressed.
+        progress (None): whether to render a progress bar (True/False), use the
+            default value ``fiftyone.config.show_progress_bars`` (None), or a
+            progress callback function to invoke instead
+    """
+    fov.validate_collection_label_fields(
+        sample_collection, in_field, fol.Detections
+    )
+
+    samples = sample_collection.select_fields(in_field)
+
+    with fou.ProgressBar(progress=progress) as pb:
+        for sample in pb(samples):
+            detections_data_json_str = sample[in_field].to_json()
```

Comment:
Can you implement this by processing `sample[in_field].detections` as a list of `Detection` objects rather than serializing to JSON? We generally try to avoid converting to other formats if not necessary.

Note that you can call `sample[in_field].copy()` to get a copy of the detections so that the existing detections aren't modified in-place

Reply:
Oh, definitely! I will check out the `.copy()` method.